### PR TITLE
Documentation for new installers API

### PR DESCRIPTION
**Pull request by timbussmann**
Adds documentation for the new installer APIs introduced via https://github.com/Particular/NServiceBus/pull/6563 in NServiceBus 8.1.

The core reference in the 8.1 csproj file is a reference to a local build since we don't have any 8.1 builds available yet.

### CODE REVIEW

File: nservicebus/operations/installers_installer-api_core_[8.1,).partial.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+When requiring special privileges to setup the endpoint, the installer API can be used to run all necessary installation steps:
+
+snippet: installer-setup
+
+When the setup requires access to services that are provided via an externally managed dependency injection container, provide the configured `IServiceProvider` to the installer API:
+
+snippet: installer-setup-externally-managed-container
+
+Note: The installer APIs always run installers, regardless whether the endpoint has configured `EnableInstallers`.
```
Comment (kbaley) [inline nservicebus/operations/installers_installer-api_core_[8.1,).partial.md:11]:
```suggestion
Note: The installer APIs always run installers, regardless of whether the endpoint has configured `EnableInstallers`.
```

File: nservicebus/operations/installers_installer-api_core_[8.1,).partial.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+The installer APIs are intended to be used when the endpoint host process be run with different privileges after installation. The `EndpointConfiguration` can't be reused in the same process to call `Endpoint.Start`. To setup the endpoint as part of the endpoint host process, use see the [Running installers during endpoint startup](#running-installers-during-endpoint-startup).
```
Comment (kbaley) [inline nservicebus/operations/installers_installer-api_core_[8.1,).partial.md:13]:
```suggestion
The installer APIs are intended to be used when resources required by the endpoint must be set up with different privileges than the endpoint itself requires. `EndpointConfiguration` can't be reused in the same process to call `Endpoint.Start`. To setup the endpoint as part of the endpoint host process, see the [Running installers during endpoint startup](#running-installers-during-endpoint-startup) article.
```

**Review by cquirosj — APPROVED** [review]

**Review by kbaley — APPROVED** [review]

**Comment by andreasohlund** [comment]
@timbussmann we're about to release 8.1, anything me and @DavidBoike can do to help get this PR ready to go?

**Comment by timbussmann** [comment]
@andreasohlund I rebased the PR, you can approve and merge it if you want.

**Review by andreasohlund — APPROVED** [review]

File: Snippets/Core/Core_8.1/Installers.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿namespace Core_8._1
```
[the remark this answers is not in the document]
Comment (timbussmann) [reply Snippets/Core/Core_8.1/Installers.cs]: that's just derived from the project name when my original csproj didn't specific a different default namespace. Fixed.

**Pull request merged**